packages/encryption/src/lib/params-validators.ts

```diff
@@ -532,43 +534,148 @@ class AccessControlConditionsValidator implements ParamsValidator {
-        errorKind: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.kind,
-        errorCode: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.name,
-      });
+      const accs = this.flattenAndFilter(
+        evmContractConditions,
+        isTokenOperator
+      );
+      for (const acc of accs) {
+        if (
+          !checkSchema(
+            acc,
+            getSchema('evmContract'),
+            'accessControlConditions',
+            this.fnName
+          )
+        ) {
+          return ELeft({
+            message:
+              'EVM Contract Access Control Conditions failed to validate the schema',
+            errorKind: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.kind,
+            errorCode: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.name,
+          });
+        }
+      }
+    }
+
+    if (solRpcConditions) {
+      if (!isValidBooleanExpression(solRpcConditions)) {
+        return ELeft({
+          message: 'Invalid boolean Solana Access Control Conditions',
+          errorKind: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.kind,
+          errorCode: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.name,
+        });
+      }
+
+      const accs = this.flattenAndFilter(solRpcConditions, isTokenOperator);
+      for (const acc of accs) {
+        if (
+          !checkSchema(
+            acc,
+            getSchema('solRpc'),
+            'accessControlConditions',
+            this.fnName
+          )
+        ) {
+          return ELeft({
+            message:
+              'Solana Access Control Conditions failed to validate the schema',
+            errorKind: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.kind,
+            errorCode: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.name,
+          });
+        }
+      }
+    }
+
+    if (unifiedAccessControlConditions) {
+      if (!isValidBooleanExpression(unifiedAccessControlConditions)) {
+        return ELeft({
+          message: 'Invalid boolean Unified Access Control Conditions',
+          errorKind: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.kind,
+          errorCode: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.name,
+        });
+      }
+
+      const accs = this.flattenAndFilter(
+        unifiedAccessControlConditions,
+        isTokenOperator
+      );
+      for (const acc of accs) {
+        if (
+          !checkSchema(
+            acc,
+            getSchema(acc.conditionType),
+            'accessControlConditions',
+            this.fnName
+          )
+        ) {
+          return ELeft({
+            message:
+              'EVM Basic Access Control Conditions failed to validate the schema',
+            errorKind: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.kind,
+            errorCode: LIT_ERROR.INVALID_BOOLEAN_EXCEPTION.name,
+          });
+        }
+      }
+    }
 
     return ERight(undefined);
   }
+
+  // Filters an array of things that can themselves be nested arrays
+  // Result is a flat array of all the things that pass the filter
+  private flattenAndFilter(
+    values: any[],
+    filter: (value: any) => boolean
+  ): any[] {
```

Comment:
can this be strongly typed?

Reply:
Typed this with a generic so we can reuse the function later

Reply:
It was removed now